src/lib/mdns/tests/TestMdnsCache.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <support/logging/CHIPLogging.h>
+// set this to 1 to enable DumpCache to see the state of the cache when needed
+// #define MDNS_LOGGING 1
+#include <iostream>
+#include <mdns/MdnsCache.h>
+
+#include <cstdint>
+#include <nlunit-test.h>
+#include <support/UnitTestRegistration.h>
+
+#include <core/CHIPError.h>
+#include <core/PeerId.h>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/InetLayer.h>
+#include <system/SystemTimer.h>
+#include <system/TimeSource.h>
+
+#define NL_TEST_DEF_FN(fn) NL_TEST_DEF("Test " #fn, fn)
+
+using namespace chip;
+using namespace chip::Mdns;
+
+void TestCreate(nlTestSuite * inSuite, void * inContext)
+{
+    MdnsCache<10> tMdnsCache;
+    tMdnsCache.DumpCache();
+}
+
+void TestInsert(nlTestSuite * inSuite, void * inContext)
+{
+    const int sizeOfCache = 5;
+    MdnsCache<sizeOfCache> tMdnsCache;
+    PeerId peerId;
+    int64_t id              = 0x100;
+    uint16_t port           = 2000;
+    Inet::InterfaceId iface = (Inet::InterfaceId) 2;
```

Comment:
please use C++ cast.

Consider using a constexpr for all these values (id/interface/port/fabricid etc.)

Reply:
Probably:

```cpp
#if CHIP_SYSTEM_CONFIG_USE_LWIP
  constexpr Inet::InterfaceId kTestInterfaceId = "xy";
#else
  constexpr Inet::InterfaceId kTestInterfaceId = 2;
#endif // CHIP_SYSTEM_CONFIG_USE_LWIP
```

Reply:
Any reason you used #if instead of #ifdef?